### PR TITLE
Menu Icon: Account for image URLs as icons.

### PR DESCRIPTION
**Pull request by obenland**
This will bring parity with WP Admin's menu, that accepts image URLs, SVGs, and dashicon slugs.

#### Changes proposed in this Pull Request

* Expands `SidebarCustomIcon` to also render image URLs in `` tags.

#### Testing instructions

* Apply D50296-code
* Visit http://calypso.localhost:3000/?flags=nav-unification and select an a8c-internal site, like your team p2.
* Make sure the VIP cloud shows up as expected.

### CODE REVIEW

**Review by cpapazoglou — APPROVED** [review]
Works as expected! Thanks!

**Comment by cpapazoglou** [comment]
Deployed r214378-wpcom

File: client/layout/sidebar/custom-icon.jsx [hunk 1]
```diff
@@ -30,12 +29,13 @@ const SidebarCustomIcon = ( { alt, className, icon, ...rest } ) => {
 			/>
 		);
 	}
-	if ( icon.indexOf( 'data:image' ) === 0 ) {
+	if ( icon.indexOf( 'data:image' ) === 0 || icon.indexOf( 'http' ) === 0 ) {
```
Comment (tyxla) [inline client/layout/sidebar/custom-icon.jsx:32]: Any chance to have `http` as part of the icon name in the future? Should we instead check if the icon starts with `http`?

Comment (cpapazoglou) [reply client/layout/sidebar/custom-icon.jsx:32]: Hey @tyxla! 
This is an effort in accepting all the icon formats wordpress core currently accepts as menu icons `URLs, SVGs, and dashicon slug`. That said, checking only `icon.indexOf( 'http' ) === 0 ` would not account for `SVGs`.

**Pull request merged**